test/Microsoft.AspNetCore.Routing.Tests/Matchers/MatcherConformanceTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public abstract class MatcherConformanceTest
+    {
+        internal abstract Matcher CreateMatcher(MatcherEndpoint endpoint);
+
+        [Fact]
+        public virtual async Task Match_SingleLiteralSegment_Success()
```

Comment:
The idea is that each matcher implementation would subclass this class. This class (the base class) contains all of the actual test definitions.